cognitedata/reveal
# improvement: Cancel prefetching of 360 images when image is removed from cache

#### Type of change

![Refactor](https://img.shields.io/badge/Type-Refactor-lightgrey) 

#### Jira ticket :blue_book:


https://cognitedata.atlassian.net/browse/REV-761

## Description :pencil:

I have added a size restriction to the `_inFlightEntities` map in `Image360LoadingCache`. If there is a new request that would exceed this limit, the oldest entity "in flight" will be aborted. 

If an event is aborted the fetch process will throw and be caught in `cachedPreload`. Here the error is logged as info. Any error that is not an abort is currently logged as a warning. 

## How has this been tested? :mag:

I have written two tests.  The first focuses on correct handling of rejected/thrown downloads and the second do checks to see that the size restriction on `_inFlightEntities` works as intended.

In addition I temporarily added logging for when images start loading, complete loading, abort and unload. I then ran Examples to see that the images were handled as expected.

## Test instructions :information_source:



I suggest setting logging level to info as this will show aborts in the console. Then load any CDF model and use the site_id: Hibernia_RS2 to load 360 images. Hover over a lot of 360 icons to see the log spam.

## Checklist :ballot_box_with_check:
